pkg/version/version.go

```diff
@@ -18,3 +19,30 @@ var (
 	// String is the human-friendly representation of the version.
 	String = fmt.Sprintf("ClusterVersionOperator %s", Raw)
 )
+
+var (
+	// commitFromGit is a constant representing the source version that
+	// generated this build. It should be set during build via -ldflags.
+	commitFromGit string
+	// versionFromGit is a constant representing the version tag that
+	// generated this build. It should be set during build via -ldflags.
+	versionFromGit string
```

Comment:
This seems like it's overlapping with out existing variables, which we set via `-ldflags` [here][1].  I'm not against pivoting to a more established approach, or borrowing a bit of tech-debt during this scaffolding phase and supporting both approaches, but I'd like to consolidate around a single approach to ingesting version information in the long term.

[1]: https://github.com/openshift/cluster-version-operator/blob/adf0cf5662854567d1ea7ce1590e0a6fed8fcaaa/hack/build-go.sh#L23-L34

Reply:
Thanks for pointing this out, I added this to make the `controllercmd.NewControllerCommandConfig` [from library-go](https://github.com/openshift/library-go/blob/master/pkg/controller/controllercmd/cmd.go#L79) happy, just needed a [apimachinery version.Info](https://github.com/kubernetes/apimachinery/blob/cfee47580787e57b3cc51782a120cf770a97316e/pkg/version/types.go#L19-L32) so I think I just copied this from some other operator, because I did not want to mess with versions in CVO - that's a downside of adding USC to CVO binary. And then I entirely forgot about it :D

Right now I'm leaning towards creating a little bit of tech debt for this, but let's make sure it is addressed before we make this TechPreview. I'll make this bit even more obvious tech debt by inlining this in USC file, not here. I'll also start sprinkling TODOs for these items (I thinkg https://github.com/openshift/cluster-version-operator/pull/1091#discussion_r1810586011 is a similar thing).